### PR TITLE
Implemented tap capability for CardPartIconLabel

### PR DESCRIPTION
**Pull request by badrinathvm**
## Issue Link :link:
 <ul>
   <li> Enhancement Feature </li>
   <li> None </li>
</ul>

## Goals of this PR :tada:
 <ul>
   <li> this provides the capability for tap functionality on CardPartIconLabel </li>
   <li> What does this fix? </li>
   <li> How far has it been tested? </li>
 </ul>
 
## How Has This Been Tested :mag:

Please let us know if you have tested your PR and if we need to reproduce the issues. Also, please let us know if we need any relevant information for running the tests.

<ul>
 <li> User Interface Testing </li>
 <li> Application Testing </li>
</ul>

## Test Configuration :space_invader:

<ul>
 <li> Xcode version: 10.2.1</li>
 <li> Device/Simulator ; iPhone Xs </li>
 <li> iOS version - 12.2 || MacOSX version</li>
</ul>

## Things to check on :dart:

 - [ ] My Pull Request code follows the coding standards and styles of the project 
 - [ ] I have worked on unit tests and reviewed my code to the best of my ability 
 - [ ] I have used comments to make other coders understand my code better 
 - [ ] My changes are good to go without any warnings 
 - [ ] I have added unit tests both for the happy and sad path 
 - [ ] All of my unit tests pass successfully before pushing the PR 
 - [ ] I have made sure all dependent downstream changes impacted by my PR are working

### CODE REVIEW

File: Example/CardParts/CardPartIconLabelCardController.swift [hunk 1]
```diff
@@ -10,6 +10,7 @@ import Foundation
 import CardParts
 import RxSwift
 import RxCocoa
+import RxGesture
```
Comment (croossin) [inline Example/CardParts/CardPartIconLabelCardController.swift]: @badrinathvm Where are we using `RxGesture`?

Comment (badrinathvm) [reply Example/CardParts/CardPartIconLabelCardController.swift]: Not required any more. removed.

File: CardParts/src/Classes/Card Parts/CardPartIconLabel.swift [hunk 2]
```diff
@@ -175,3 +177,53 @@ extension Reactive where Base: CardPartIconLabel {
...
+        self.isUserInteractionEnabled = true
+        self.tapGestureRecognizerAction = action
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleTapGesture))
+        self.addGestureRecognizer(tapGestureRecognizer)
+    }
+
+    // Every time the user taps on the UIImageView, this function gets called, which triggers the closure we stored
+    @objc fileprivate func handleTapGesture(sender: UITapGestureRecognizer) {
+        let iconViewFrame = (sender.view as? CardPartIconLabel)?.iconView?.frame
+        if let action = self.tapGestureRecognizerAction {
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartIconLabel.swift]: How about one guard let?

`guard let action = self.tapGestureRecognizerAction, 
let iconWidth = iconViewFrame?.size.width, 
let originX = iconViewFrame?.origin.x, 
iconViewPosition.x <= ( iconWidth + originX ) else { return }`

Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartIconLabel.swift]: done.

**Pull request merged**